app/ui/legacy/src/main/java/com/fsck/k9/activity/LauncherShortcuts.kt

```diff
@@ -0,0 +1,56 @@
+package com.fsck.k9.activity
+
+import android.content.Intent
+import android.os.Bundle
+import com.fsck.k9.BaseAccount
+import com.fsck.k9.activity.MessageList.Companion.shortcutIntent
+import com.fsck.k9.activity.MessageList.Companion.shortcutIntentForAccount
+import com.fsck.k9.search.SearchAccount
+import com.fsck.k9.ui.R
+
+class LauncherShortcuts : AccountList() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setTitle(R.string.shortcuts_title)
+
+        // finish() immediately if we aren't supposed to be here
+        if (Intent.ACTION_CREATE_SHORTCUT != intent.action) {
+            finish()
+        }
+    }
+
+    override fun onAccountSelected(account: BaseAccount) {
+        val shortcutIntent: Intent
+        if (account is SearchAccount) {
+            shortcutIntent = shortcutIntent(this, account.id)
+        } else {
+            shortcutIntent = shortcutIntentForAccount(this, account.uuid)
+        }
```

Comment:
Optional suggestions: I think prefixing the methods with `MessageList.` makes it more obvious where the data is coming from.

```suggestion
        val shortcutIntent = if (account is SearchAccount) {
            MessageList.shortcutIntent(this, account.id)
        } else {
            MessageList.shortcutIntentForAccount(this, account.uuid)
        }
```